### PR TITLE
add nutanix upgrade pre-steps

### PR DESCRIPTION
**Pull request by huali9**
This is for https://issues.redhat.com/browse/OCPQE-15938 @sunzhaohua2 @miyadav @jhou1 PTAL, thanks!
Upgrade succeed. https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/41579/rehearse-41579-periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-stable-4.12-nutanix-ipi-disconnected-p2-f28/1683418047068508160

### CODE REVIEW

**Comment by huali9** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-stable-4.12-nutanix-ipi-disconnected-p2-f28

**Comment by huali9** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-stable-4.12-nutanix-ipi-disconnected-p2-f28

**Comment by sunzhaohua2** [comment]
Thanks @huali9 looks good to me, I think we can update step name to nutanix-upgrade-procedure and update all 4.12-4.13 nutanix jobs to include this step.

**Comment by huali9** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-stable-4.13-upgrade-from-stable-4.12-nutanix-ipi-proxy-fips-p2-f14

**Comment by huali9** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-stable-4.12-nutanix-ipi-disconnected-p2-f28

**Comment by huali9** [comment]
> Thanks @huali9 looks good to me, I think we can update step name to nutanix-upgrade-procedure and update all 4.12-4.13 nutanix jobs to include this step.

Thanks @sunzhaohua2 updated.
new upgrade job succeed. https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/41579/rehearse-41579-periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-stable-4.13-upgrade-from-stable-4.12-nutanix-ipi-proxy-fips-p2-f14/1683726808584818688

File: ci-operator/step-registry/nutanix/upgrade/procedure/nutanix-upgrade-procedure-ref.metadata.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+	"path": "nutanix/upgrade/procedure/nutanix-upgrade-procedure-ref.yaml",
```
Comment (jianlinliu) [inline ci-operator/step-registry/nutanix/upgrade/procedure/nutanix-upgrade-procedure-ref.metadata.json:2]: how about renaming `procedure` to `creds`, or `pre-config`

Comment (huali9) [reply ci-operator/step-registry/nutanix/upgrade/procedure/nutanix-upgrade-procedure-ref.metadata.json:2]: I'm really bad at naming things, `procedure` was suggested by Zhaohua https://github.com/openshift/release/pull/41579#issuecomment-1648972947, I'd like to get your opinions of the final name. `procedure`, `creds`, or `pre-config`? @jianlinliu @sunzhaohua2

Comment (jianlinliu) [reply ci-operator/step-registry/nutanix/upgrade/procedure/nutanix-upgrade-procedure-ref.metadata.json:2]: I prefer pre-config, @sunzhaohua2 wdyt?

Comment (sunzhaohua2) [reply ci-operator/step-registry/nutanix/upgrade/procedure/nutanix-upgrade-procedure-ref.metadata.json:2]: ok to me

Comment (huali9) [reply ci-operator/step-registry/nutanix/upgrade/procedure/nutanix-upgrade-procedure-ref.metadata.json:2]: Thank you @jianlinliu @sunzhaohua2 Let me update the name.

**Comment by jianlinliu** [comment]
cc @liangxia to review, cc @shellyyang1989 and @jiajliu

**Comment by shellyyang1989** [comment]
Thank you Huali! Just want to confirm it's only required on the 4.12 to 4.13 upgrade, right?

**Comment by huali9** [comment]
> Thank you Huali! Just want to confirm it's only required on the 4.12 to 4.13 upgrade, right?

Yes, it's only required on the 4.12 to 4.13 upgrade on Nutanix.

**Pull request merged**